src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/WrappedGr.hpp

```diff
@@ -0,0 +1,156 @@
+class DataVector;
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+namespace GeneralizedHarmonic {
+namespace Solutions {
+
+/*!
+ * \brief A wrapper for general-relativity analytic solutions that loads
+ * the analytic solution and then adds a function that returns
+ * any combination of the generalized-harmonic evolution variables,
+ * specifically `gr::Tags::SpacetimeMetric`, `GeneralizedHarmonic::Tags::Pi`,
+ * and `GeneralizedHarmonic::Tags::Phi`
+ */
+template <typename SolutionType>
+class WrappedGr : public SolutionType {
+ public:
+  using SolutionType::SolutionType;
+
+  static constexpr size_t volume_dim = SolutionType::volume_dim;
+  using options = typename SolutionType::options;
+  static constexpr OptionString help = SolutionType::help;
+
+  using DerivLapse = ::Tags::deriv<gr::Tags::Lapse<DataVector>,
+                                   tmpl::size_t<volume_dim>, Frame::Inertial>;
+  using DerivShift =
+      ::Tags::deriv<gr::Tags::Shift<volume_dim, Frame::Inertial, DataVector>,
+                    tmpl::size_t<volume_dim>, Frame::Inertial>;
+  using DerivSpatialMetric = ::Tags::deriv<
+      gr::Tags::SpatialMetric<volume_dim, Frame::Inertial, DataVector>,
+      tmpl::size_t<volume_dim>, Frame::Inertial>;
+  using TimeDerivLapse = ::Tags::dt<gr::Tags::Lapse<DataVector>>;
+  using TimeDerivShift =
+      ::Tags::dt<gr::Tags::Shift<volume_dim, Frame::Inertial, DataVector>>;
+  using TimeDerivSpatialMetric = ::Tags::dt<
+      gr::Tags::SpatialMetric<volume_dim, Frame::Inertial, DataVector>>;
+
+  using IntermediateVars = tuples::tagged_tuple_from_typelist<
+      typename SolutionType::template tags<DataVector>>;
+
+  template <typename... Tags>
+  tuples::TaggedTuple<Tags...> variables(
+      const tnsr::I<DataVector, volume_dim>& x, double t,
+      tmpl::list<Tags...> /*meta*/) const noexcept {
+    // Get the underlying solution's variables using the solution's tags list,
+    // store in IntermediateVariables
+    const IntermediateVars& intermediate_vars = SolutionType::variables(
+        x, t, typename SolutionType::template tags<DataVector>{});
+
+    return {
+        get<Tags>(variables(x, t, tmpl::list<Tags>{}, intermediate_vars))...};
+  }
+
+  template <typename Tag>
+  tuples::TaggedTuple<Tag> variables(const tnsr::I<DataVector, volume_dim>& x,
+                                     double t, tmpl::list<Tag> /*meta*/) const
+      noexcept {
+    const IntermediateVars& intermediate_vars = SolutionType::variables(
+        x, t, typename SolutionType::template tags<DataVector>{});
+    return {get<Tag>(variables(x, t, tmpl::list<Tag>{}, intermediate_vars))};
+  }
+
+  // clang-tidy: google-runtime-references
+  void pup(PUP::er& p) noexcept { SolutionType::pup(p); }  // NOLINT
+
+ private:
+  // Preprocessor logic to avoid declaring variables() functions for
+  // tags other than the three the wrapper adds (i.e., other than
+  // gr::Tags::SpacetimeMetric, gr::Tags::Lapse, and gr::Tags::Shift.
```

Comment:
Shouldn't `Lapse` and `Shift` be `GeneralizedHarmonic::Tags::Phi` and `GeneralizedHarmonic::Tags::Pi`? And you're missing a closing paren :)

Reply:
Yes, my brain made a caching error :) Fixed and squashed